File: Backend/Microservices/Bursary/Eefa.Bursary.Application/UseCases/Definitions/BankAccount/Commands/Update/UpdateBankAccountCommand.cs
using AutoMapper;
using Eefa.Bursary.Application.UseCases.Definitions.BankAccount.Commands.Add;
using Eefa.Bursary.Domain.Entities.Definitions;
using Eefa.Bursary.Infrastructure.Interfaces;
using Eefa.Common;
using Eefa.Common.CommandQuery;
using Eefa.Common.Exceptions;
using MediatR;
using Microsoft.EntityFrameworkCore;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Eefa.Bursary.Application.UseCases.Definitions.BankAccount.Commands.Update
{
    public class UpdateBankAccountCommand : CommandBase, IRequest<ServiceResult<BankAccounts>>, IMapFrom<UpdateBankAccountCommand>, ICommand
    {
        public int? Id { get; set; }
        public int? ParentId { get; set; }
        public int BankBranchId { get; set; } = default!;
        public string Sheba { get; set; } = default!;
        public string AccountNumber { get; set; } = default!;
        public int? SubsidiaryCodeId { get; set; }
        public int? RelatedBankAccountId { get; set; }
        public int AccountTypeBaseId { get; set; } = default!;
        public int AccountHeadId { get; set; }
        public int? AccountReferencesGroupId { get; set; }
        public int? ReferenceId { get; set; }
        public int? AccountStatus { get; set; }
        public decimal WithdrawalLimit { get; set; } = 0;
        public bool HaveChekBook { get; set; } = default!;
        public int CurrenceTypeBaseId { get; set; } = default!;
        public string SignersJson { get; set; } = "";

        public void Mapping(Profile profile)
        {
            profile.CreateMap<UpdateBankAccountCommand, BankAccounts>().IgnoreAllNonExisting();
        }
    }

    public class UpdateBankAccountCommandHandler : IRequestHandler<UpdateBankAccountCommand, ServiceResult<BankAccounts>>
    {
        private readonly IMapper _mapper;
        private readonly IBursaryUnitOfWork _uow;

        public UpdateBankAccountCommandHandler(IMapper mapper, IBursaryUnitOfWork uow)
        {
            _mapper = mapper;
            _uow = uow;
        }

        public async Task<ServiceResult<BankAccounts>> Handle(UpdateBankAccountCommand request, CancellationToken cancellationToken)
        {
            var bankaccount = await _uow.BankAccounts.FirstOrDefaultAsync(x => x.Id == request.Id);
            if (bankaccount == null)
            {
                throw new ValidationError("اطلاعات ورودی ارسال نگردیده است");
            }
            _mapper.Map(request, bankaccount);

            _uow.BankAccounts.Update(bankaccount);
            var value = await _uow.SaveChangesAsync(cancellationToken);

            if (value <= 0)
                throw new Exception("بروز خطا در ثبت اطلاعات بانک");
            return ServiceResult<BankAccounts>.Success(bankaccount);
        }
    }


}

File: Frontend/angular-app/src/app/modules/bursary/repositories/bank-account/commands/update-bank-account-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {BankAccount} from "../../../entities/bank-account";

export class UpdateBankAccountCommand extends IRequest<UpdateBankAccountCommand, BankAccount> {
  public id:number | undefined = undefined;
  parentId: number | undefined = undefined;
  bankBranchId: number | undefined = undefined;
  bankBranchTitle: string | undefined = undefined;
  title: string | undefined = undefined;
  sheba: string | undefined = undefined;
  accountNumber: string | undefined = undefined;
  subsidiaryCodeId: number | undefined = undefined;
  relatedBankAccountId: number | undefined = undefined;
  accountTypeBaseId: number | undefined = undefined;
  accountStatus: number | undefined = undefined;
  withdrawalLimit: number | undefined = undefined;
  haveChekBook: boolean | undefined = undefined;
  currenceTypeBaseId: number | undefined = undefined;
  signersJson: string | undefined = undefined;


  accountReferenceId: number | undefined = undefined;
  accountReferenceTitle: number | undefined = undefined;
  accountReferenceCode: number | undefined = undefined;
  constructor() {
    super();
  }

  mapFrom(entity: BankAccount): UpdateBankAccountCommand {
    this.mapBasics(entity, this);
    return this;
  }

  mapTo(): BankAccount {
    throw new ApplicationError(UpdateBankAccountCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/bursary/bankAccounts/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdateBankAccountCommandHandler.name)
export class UpdateBankAccountCommandHandler implements IRequestHandler<UpdateBankAccountCommand, BankAccount> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService
  ) {
  }

  async Handle(request: UpdateBankAccountCommand): Promise<BankAccount> {
    let httpRequest: HttpRequest<UpdateBankAccountCommand> = new HttpRequest<UpdateBankAccountCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Put<UpdateBankAccountCommand, ServiceResult<BankAccount>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })
  }
}
